lib/backends/sdc/volumes.js

```diff
@@ -109,7 +119,19 @@ function createVolume(volumeParams, options, callback) {
 
                 ctx.volume = volume;
                 pollVolumeState(volume.uuid, volapiClient,
-                    ['ready', 'failed'], done);
+                    ['ready', 'failed'],
+                    function _onPollComplete(err, vol) {
+                        var createErr = err;
+
+                        if (!err && vol.state === 'failed') {
```

Comment:
Check that vol is not undefined with if (!err && vol && vol.state === 'failed'), or maybe assert that vol is not undefined when err is falsy.